rust/azure_iot_operations_connector/src/file_mount_azure_device_registry.rs

```diff
@@ -1,4 +1,278 @@
+    pub async fn observe_asset_delete(
+        &self,
+        _asset_ref: AssetRef,
+        _timeout: Duration,
+    ) -> Result<AssetDeleteObservation, FileMountError> {
+        let () = tokio::task::yield_now().await;
+        Ok(AssetDeleteObservation {
+            receiver: tokio::sync::oneshot::channel().1,
+        })
+    }
+}
+
+/// Represents an observation for device endpoint creation events.
+///
+/// This struct contains an internal channel for receiving notifications
+/// about newly created device endpoints.
+pub struct DeviceEndpointCreateObservation {
+    receiver: UnboundedReceiver<DeviceEndpointRef>,
+}
+
+impl DeviceEndpointCreateObservation {
+    /// Receives a notification for a newly created device endpoint.
+    ///
+    /// # Returns
+    /// An `Option` containing a `DeviceEndpointRef` if a notification is received, or `None` if the channel is closed.
+    pub async fn recv_notification(&mut self) -> Option<DeviceEndpointRef> {
+        self.receiver.recv().await
+    }
+}
+
+/// Represents an observation for device endpoint deletion events.
+///
+/// This struct contains an internal channel for receiving notifications
+/// about deleted device endpoints.
+pub struct DeviceEndpointDeleteObservation {
+    /// The internal channel for receiving notifications for an device deletion event.
+    #[allow(dead_code)]
+    receiver: oneshot::Receiver<DeviceEndpointRef>,
+}
+
+impl DeviceEndpointDeleteObservation {
+    /// Receives a notification for a deleted device endpoint.
+    ///
+    /// # Returns
+    /// An `Option` containing a `DeviceEndpointRef` if a notification is received, or `None` if the channel is closed.
+    pub async fn recv_notification(self) -> Option<DeviceEndpointRef> {
+        self.receiver.await.ok()
+    }
+}
+
+/// Represents a device and its associated endpoint.
+pub struct DeviceEndpointRef {
+    /// The name of the device
+    pub device_name: String,
+    /// The name of the endpoint
+    pub endpoint_name: String,
+}
+
+/// Represents an observation for asset creation events.
+///
+/// This struct contains an internal channel for receiving notifications
+/// about newly created assets.
+pub struct AssetCreateObservation {
+    /// The internal channel for receiving notifications for an asset creation event.
+    #[allow(dead_code)]
+    receiver: UnboundedReceiver<AssetRef>,
+}
+
+impl AssetCreateObservation {
+    /// Receives a notification for a newly created asset.
+    ///
+    /// # Returns
+    /// An `Option` containing an `AssetRef` if a notification is received, or `None` if the channel is closed.
+    pub async fn recv_notification(&mut self) -> Option<AssetRef> {
+        self.receiver.recv().await
+    }
+}
+
+/// Represents an observation for asset deletion events.
+///
+/// This struct contains an internal channel for receiving notifications
+/// about deleted assets.
+pub struct AssetDeleteObservation {
```

Comment:
need impl for this, and then can probably remove the dead_code allow for both delete receivers too